components/server/src/server.rs

```diff
@@ -1472,14 +1479,47 @@ where
                     .get_engine_size()
                     .expect("get raft engine size");
 
+                let mut raft_disk_status = disk::DiskUsage::Normal;
```

Comment:
I think we can add the failpoint here so that we can set the raft_disk_status to whatever we like, and then can follow the test code in tests/failpoints/cases/test_disk_full.rs. We just need one integration test to ensure the functionality works.

Reply:
Integration test could not run into `fn init_storage_stats_task` in which you marked here, so it's may not easy to have an integration case for the case. Fortunately,  the manual test should cover the code path.